src/main/java/org/jabref/gui/dialogs/BackupUIManager.java

```diff
@@ -17,25 +29,45 @@ public class BackupUIManager {
-                .toString();
-
-        boolean restoreClicked = DefaultTaskExecutor.runInJavaFXThread(() -> dialogService.showConfirmationDialogAndWait(
-                Localization.lang("Backup found"), content,
-                Localization.lang("Restore from backup"),
-                Localization.lang("Ignore backup")));
-
-        if (restoreClicked) {
-            BackupManager.restoreBackup(originalPath);
+    public static Optional<ParserResult> showRestoreBackupDialog(DialogService dialogService, Path originalPath) {
+        var actionOpt = showBackupResolverDialog(dialogService, originalPath);
+        return actionOpt.flatMap(action -> {
+            if (action == BackupResolverDialog.RESTORE_FROM_BACKUP) {
+                BackupManager.restoreBackup(originalPath);
+                return Optional.empty();
+            } else if (action == BackupResolverDialog.REVIEW_BACKUP) {
+                return showReviewBackupDialog(dialogService, originalPath);
+            }
+            return Optional.empty();
+        });
+    }
+
+    private static Optional<ButtonType> showBackupResolverDialog(DialogService dialogService, Path originalPath) {
+        return DefaultTaskExecutor.runInJavaFXThread(() -> dialogService.showCustomDialogAndWait(new BackupResolverDialog(originalPath)));
+    }
+
+    private static Optional<ParserResult> showReviewBackupDialog(DialogService dialogService, Path originalPath) {
+        try {
+            Path backupPath = BackupFileUtil.getPathOfLatestExisingBackupFile(originalPath, BackupFileType.BACKUP).orElseThrow();
+            ImportFormatPreferences importFormatPreferences = Globals.prefs.getImportFormatPreferences();
+            ParserResult originalParserResult = OpenDatabase.loadDatabase(originalPath, importFormatPreferences, new DummyFileUpdateMonitor());
+            BibDatabaseContext originalDatabase = originalParserResult.getDatabaseContext();
+            BibDatabaseContext backupDatabase = OpenDatabase.loadDatabase(backupPath, importFormatPreferences, new DummyFileUpdateMonitor()).getDatabaseContext();
+
+            return DefaultTaskExecutor.runInJavaFXThread(() -> {
+                DatabaseChangesResolverDialog reviewBackupDialog = new DatabaseChangesResolverDialog(
+                        DatabaseChangeList.compareAndGetChanges(originalDatabase, backupDatabase, null),
+                        originalDatabase, dialogService, Globals.stateManager, Globals.getThemeManager(), Globals.prefs, "Review Backup"
+                );
+                var allChangesResolved = dialogService.showCustomDialogAndWait(reviewBackupDialog);
+                if (allChangesResolved.isEmpty() || !allChangesResolved.get()) {
+                    return showRestoreBackupDialog(dialogService, originalPath);
+                }
+                return Optional.of(originalParserResult);
+            });
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Please wrap this in a a logger....

Reply:
Sorry for the delay, I was a little busy this week.